plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/ReadSessionCreator.java

```diff
@@ -63,36 +63,28 @@ public ReadSessionCreator(
                 .filter(BigQueryUtil::validColumnName)
                 .collect(toList());
 
-        try (BigQueryStorageClient bigQueryStorageClient = bigQueryStorageClientFactory.createBigQueryStorageClient()) {
-            ReadOptions.TableReadOptions.Builder readOptions = ReadOptions.TableReadOptions.newBuilder()
+        try (BigQueryReadClient bigQueryReadClient = bigQueryReadClientFactory.createBigQueryReadClient()) {
+            ReadSession.TableReadOptions.Builder readOptions = ReadSession.TableReadOptions.newBuilder()
                     .addAllSelectedFields(filteredSelectedFields);
             filter.ifPresent(readOptions::setRowRestriction);
 
-            TableReferenceProto.TableReference tableReference = toTableReference(actualTable.getTableId());
-
-            Storage.ReadSession readSession = bigQueryStorageClient.createReadSession(
-                    Storage.CreateReadSessionRequest.newBuilder()
+            ReadSession readSession = bigQueryReadClient.createReadSession(
+                    CreateReadSessionRequest.newBuilder()
                             .setParent("projects/" + bigQueryClient.getProjectId())
-                            .setFormat(Storage.DataFormat.AVRO)
-                            .setRequestedStreams(parallelism)
-                            .setReadOptions(readOptions)
-                            .setTableReference(tableReference)
-                            // The BALANCED sharding strategy causes the server to
-                            // assign roughly the same number of rows to each stream.
-                            .setShardingStrategy(Storage.ShardingStrategy.BALANCED)
+                            .setReadSession(ReadSession.newBuilder()
+                                    .setDataFormat(DataFormat.AVRO)
+                                    .setTable(toTableResourceName(actualTable.getTableId()))
+                                    .setReadOptions(readOptions))
+                            .setMaxStreamCount(parallelism)
```

Comment:
maybe not for this PR but how is parallelism determined?  Should we let the service determine this?

Reply:
I agree this is an area where we might want to investigate further but I think it's out of scope for this CL, which is just about keeping the lights on.

Reply:
The default paralellism is equal to number of workers in cluster. It can be controlled using `bigquery.paralellism` catalog config property.